lib/atlas/active_document/serialized_graph_validator.rb

```diff
@@ -0,0 +1,67 @@
+module Atlas
+  module ActiveDocument
+    class SerializedGraphValidator < ActiveModel::Validator
+      def validate(record)
+        return false if skip_validations?(record)
+
+        @graph = GraphDeserializer.new(YAML.load_file(record.graph_path))
```

Comment:
Using instance variables in validators isn't thread-safe; the same instance is re-used each time a record is validated. Instead, use a normal variable and pass it explicitly to the other `validate_*` methods:

```ruby
graph = GraphDeserializer.new(...)

validate_nodes(record, graph)
validate_edges(record, graph)
validate_slots(record, graph)
```